File: JULY-2025/04-07-2025/p3307.php
class Solution {

    /**
     * @param Integer $k
     * @param Integer[] $operations
     * @return String
     */
        function kthCharacter(int $k, array $operations): string {
            $countOps = 0;
            $val = $k;
            while ($val > 1) {
                $jumps = (int)ceil(log($val, 2));
                $val -= 1 << ($jumps - 1);
                $countOps += $operations[$jumps - 1];
            }
            $ch = chr(ord('a') + ($countOps % 26));
            return $ch;
        }
}

File: JULY-2025/04-07-2025/p3307.rb
# @param {Integer} k
# @param {Integer[]} operations
# @return {Character}
def kth_character(k, operations)
  count = 0
  while k > 1
    pos = Math.log2(k).floor
    if k & (k - 1) == 0
      count += operations[0...pos].sum
      break
    else
      count += operations[pos]
      k -= 2**pos
    end
  end
  ('a'.ord + count % 26).chr
end